src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -895,6 +876,15 @@ protected NodeModel()
             ArgumentLacing = LacingStrategy.Disabled;
 
             RaisesModificationEvents = true;
+
+            InPorts.CollectionChanged += PortsCollectionChanged;
+            OutPorts.CollectionChanged += PortsCollectionChanged;
+        }
+
+        private void PortsCollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            ValidateConnections();
+            ConfigureSnapEdges(sender == InPorts?InPorts : OutPorts);
```

Comment:
Yes. The `CollectionChanged` event's `sender` is the collection. You can get at the elements that have been changed by using the [`NotifyCollectionChangedEventArgs`](https://msdn.microsoft.com/en-us/library/system.collections.specialized.notifycollectionchangedeventargs(v=vs.110).aspx).